framework/physics/FieldFunction/fieldfunction_gridbased.h

```diff
@@ -67,36 +58,53 @@ class FieldFunctionGridBased : public FieldFunction
 
   virtual ~FieldFunctionGridBased() = default;
 
-private:
-  VectorGhostCommPtr MakeGhostCommunicator();
-
-public:
   // Getters
+  /**Returns the spatial discretization method.*/
   const chi_math::SpatialDiscretization& SDM() const { return *sdm_; }
-  const std::vector<double>& FieldVectorRead() const { return field_vector_; }
-  std::vector<double>& FieldVector() { return field_vector_; }
+  /**Returns a read-only reference to the locally stored field data.*/
+  const std::vector<double>& FieldVectorRead() const;
+  /**Returns a reference to the locally stored field data.*/
+  std::vector<double>& FieldVector();
 
   // 01 Updates
+  /**Updates the field vector with a local STL vector.*/
   void UpdateFieldVector(const std::vector<double>& field_vector);
+  /**Updates the field vector with a PETSc vector. This only operates locally.*/
   void UpdateFieldVector(const Vec& field_vector);
 
   // 03 Export VTK
+  /**Static method to export multiple grid-based field functions.*/
   typedef std::vector<std::shared_ptr<const FieldFunctionGridBased>> FFList;
   static void ExportMultipleToVTK(const std::string& file_base_name,
                                   const FFList& ff_list);
 
-public:
   // 04 Utils
+  /**Makes a copy of the locally stored data with ghost access.*/
   std::vector<double> GetGhostedFieldVector() const;
 
   // 05 Point Values
   /**\brief Returns the component values at requested point.*/
   virtual std::vector<double>
   GetPointValue(const chi_mesh::Vector3& point) const;
 
+  /**Evaluates the field function, on a cell, at the specified point.*/
   double Evaluate(const chi_mesh::Cell& cell,
                   const chi_mesh::Vector3& position,
                   unsigned int component) const override;
+
+protected:
+  chi_math::SMDPtr sdm_;
```

Comment:
Out of scope, but the type should be `SDMPtr` not `SMDPtr`. Commenting just to put this on the radar. We should move away from calling a discretization an SDM=SpatialDiscretizationMethod. Method implies a type, not an object (i.e. like an enum class).

Reply:
Whoops. I can read that 10 times and not see it. Done